### ISSUE TITLE
Scala 2.12.8 Play 2.7 not working with plugin

### ISSUE THREAD

**Issue by mihaisoloi** [iheartradio/play-swagger#239]
If you try running with the plug-in installed on Scala 2.12.8 Play 2.7 you get this stack trace and the process stops.

```
An exception or error caused a run to abort: play.api.libs.json.Reads$.traversableReads(Lscala/collection/generic/CanBuildFrom;Lplay/api/libs/json/Reads;)Lplay/api/libs/json/Reads; 
java.lang.NoSuchMethodError: play.api.libs.json.Reads$.traversableReads(Lscala/collection/generic/CanBuildFrom;Lplay/api/libs/json/Reads;)Lplay/api/libs/json/Reads;
	at com.iheart.playSwagger.Domain$CustomTypeMapping$.<init>(Domain.scala:59)
	at com.iheart.playSwagger.Domain$CustomTypeMapping$.<clinit>(Domain.scala)
	at com.iheart.playSwagger.SwaggerSpecGenerator.customMappings$lzycompute(SwaggerSpecGenerator.scala:260)
	at com.iheart.playSwagger.SwaggerSpecGenerator.customMappings(SwaggerSpecGenerator.scala:259)
	at com.iheart.playSwagger.SwaggerSpecGenerator.$anonfun$endPointSpec$4(SwaggerSpecGenerator.scala:370)
	at scala.collection.TraversableLike$WithFilter.$anonfun$map$2(TraversableLike.scala:742)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:741)
	at com.iheart.playSwagger.SwaggerSpecGenerator.$anonfun$endPointSpec$2(SwaggerSpecGenerator.scala:368)
	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:244)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:244)
	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:241)
	at scala.collection.immutable.List.flatMap(List.scala:355)
	at com.iheart.playSwagger.SwaggerSpecGenerator.endPointSpec(SwaggerSpecGenerator.scala:367)
	at com.iheart.playSwagger.SwaggerSpecGenerator.endPointEntry(SwaggerSpecGenerator.scala:333)
	at com.iheart.playSwagger.SwaggerSpecGenerator.$anonfun$paths$1(SwaggerSpecGenerator.scala:309)
	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:244)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:244)
	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:241)
	at scala.collection.immutable.List.flatMap(List.scala:355)
	at com.iheart.playSwagger.SwaggerSpecGenerator.paths(SwaggerSpecGenerator.scala:309)
	at com.iheart.playSwagger.SwaggerSpecGenerator.$anonfun$generateFromRoutes$1(SwaggerSpecGenerator.scala:124)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:237)
	at scala.collection.Iterator.foreach(Iterator.scala:941)
	at scala.collection.Iterator.foreach$(Iterator.scala:941)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:237)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:230)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at com.iheart.playSwagger.SwaggerSpecGenerator.generateFromRoutes(SwaggerSpecGenerator.scala:121)
	at com.iheart.playSwagger.SwaggerSpecGenerator.$anonfun$generateFromRoutesFile$16(SwaggerSpecGenerator.scala:106)
	at scala.util.Success.flatMap(Try.scala:251)
	at com.iheart.playSwagger.SwaggerSpecGenerator.generateFromRoutesFile(SwaggerSpecGenerator.scala:105)
	at com.iheart.playSwagger.SwaggerSpecGenerator.generate(SwaggerSpecGenerator.scala:50)
```

**Comment by iomonad** [comment]
Same issue occurred when upgrading: ```2.7.0-M3``` to ```2.7.0```.
Produce the same exception: 
```
(run-main-0) java.lang.NoSuchMethodError: play.api.libs.json.Reads$. 
(...)
```

**Comment by kailuowang** [comment]
This is a binary compatibility issue. We need to compile play-swagger against play 2.7. I just created a play2.7, if anyone interested in upgrade play to 2.7 and PR to that branch I'll be happy to release from it.

**Comment by mihaisoloi** [comment]
@kailuowang I was trying to do it, but apparently in the upgrade we have a failing test in `SwaggerSpecGeneratorIntegrationSpec` called `should maintain route file order`. It seems that the routes get scrambled up now, I will take a closer look tonight

**Comment by kailuowang** [comment]
Just released 0.7.5-PLAY2.7 (thanks to @mihaisoloi 's contribution) which should solve this issue.

**Comment by mihaisoloi** [comment]
As this is merged, deployed, and tested as I've just deployed the Play app we have with 2.7 it's safe to say we can close it.

One thing to add though, because `play-json` 2.7.0 is broken you need to add a library dependency for `"com.typesafe.play"                %% "play-json"                % "2.7.1",` to your Play project.

**Comment by fralken** [comment]
Hello, you should keep this issue open (as a remainder for others) because with play-swagger 0.7.5-PLAY2.7 a Play 2.7 project doesn't build unless you explicitly add a dependency to play-json 2.7.1, since Play 2.7.0 depends on play-json 2.7.0.

I woked around this by adding:
```
dependencyOverrides ++= Seq(
  "com.typesafe.play" %% "play-json" % "2.7.1"
)
```

in `build.sbt`. Probably this will be automatically fixed when Play 2.7.1 will be released. 
Thanks
